src/lib/operators/aggregate_sort.cpp

```diff
@@ -0,0 +1,585 @@
+                             const std::vector<AggregateColumnDefinition>& aggregates,
+                             const std::vector<ColumnID>& groupby_column_ids)
+    : AbstractAggregateOperator(in, aggregates, groupby_column_ids) {}
+
+const std::string AggregateSort::name() const { return "AggregateSort"; }
+
+/**
+ * Calculates the value for the <code>aggregate_index</code>'th aggregate.
+ * To do this, we iterate over all segments of the corresponding column.
+ * The aggregate values (one per group-by-combination) are built incrementally.
+ * Every time we reach the begin of a new group-by-combination,
+ * we store the aggregate value for the current (now previous) group.
+ *
+ * @tparam ColumnType the type of the input column to aggregate on
+ * @tparam AggregateType the type of the aggregate (=output column)
+ * @tparam function the aggregate function, as type parameter - e.g. AggregateFunction::MIN, AVG, COUNT, ...
+ * @param aggregate_group_pointers the row ids where a new combination in the sorted table begins
+ * @param aggregate_index determines which aggregate to calculate (from _aggregates)
+ * @param aggregate_function the aggregate function - as callable object. This should always be the same as <code>function</code>
+ * @param sorted_table the input table, sorted by the group by columns
+ */
+template <typename ColumnType, typename AggregateType, AggregateFunction function>
+void AggregateSort::_aggregate_values(const std::set<RowID>& aggregate_group_pointers, const uint64_t aggregate_index,
+                                      const std::shared_ptr<const Table>& sorted_table) {
+  auto aggregate_function = AggregateFunctionBuilder<ColumnType, AggregateType, function>().get_aggregate_function();
+
+  // We already know beforehand how many aggregate values (=group-by-combinations) we have to calculate
+  const size_t num_groups = aggregate_group_pointers.size() + 1;
+
+  // Vectors to store aggregate values (and if they are NULL) for later usage in value segments
+  auto aggregate_results = std::vector<AggregateType>(num_groups);
+  auto aggregate_null_values = std::vector<bool>(num_groups, false);
+
+  // Variables needed for the aggregates. Not all variables are needed for all aggregates
+
+  // Row counts per group, ex- and including null values. Needed for count (<column>/*) and average
+  uint64_t value_count = 0u;
+  uint64_t value_count_with_null = 0u;
+
+  // All unique values found. Needed for count distinct
+  std::unordered_set<ColumnType> unique_values;
+
+  // The number of the current group-by-combination. Used as offset when storing values
+  uint64_t aggregate_group_index = 0u;
+
+  const auto& chunks = sorted_table->chunks();
+
+  std::optional<AggregateType> current_aggregate_value;
+  ChunkID current_chunk_id{0};
+  if (function == AggregateFunction::Count && !_aggregates[aggregate_index].column) {
+    /*
+     * Special COUNT(*) implementation.
```

Comment:
Which test covers this?

Reply:
This is covered by the [`(One|Two|Three)GroupbyCountStar` tests](https://github.com/ekrebs5/hyrise/blob/aggregate-benchmark/src/test/operators/aggregate_test.cpp#L476).

Reply:
@mrks resolve?